prometheus/http.go

```diff
@@ -15,14 +15,169 @@ package prometheus
 
 import (
 	"bufio"
+	"compress/gzip"
+	"fmt"
 	"io"
+	"log"
 	"net"
 	"net/http"
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+const (
+	contentTypeHeader     = "Content-Type"
+	contentLengthHeader   = "Content-Length"
+	contentEncodingHeader = "Content-Encoding"
+	acceptEncodingHeader  = "Accept-Encoding"
+)
+
+// Handler returns an HTTP handler for the DefaultRegistry. It is
+// already instrumented with InstrumentHandler (using "prometheus" as handler
+// name).
+//
+// Please note the issues described in the doc comment of InstrumentHandler. You
+// might want to consider using UninstrumentedHandler instead.
+//
+// The returned Handler is using the same HandlerOpts as the Handler returned by
+// UninstrumentedHandler. See there for details.
+func Handler() http.Handler {
```

Comment:
> That's a purely formal argument. Sticking functions to structs is
sugarcoating and not reducing the mental load. But I said that again.

It is reducing the mental load, as when e.g. I'm looking for functions related to the Registry they're right there as methods rather than spread around the package.

> Packages are the Go way of grouping related functions.

Classes are a Go way to group functions related to a specific type.

> I don't think it's a good idea to coerce the Go API into that of a very different language.

Considering the languages we have clients in, it's actually pretty similar (you could make all the above arguments about Java for example). The only real major difference API-wise is that functions can only have one signature. I'd also expect Ruby and Python to look roughly the same as each other.